cfme/cloud/instance/azure.py

```diff
@@ -78,17 +64,22 @@ def power_control_from_provider(self, option):
         else:
             raise OptionNotAvailable(option + " is not a supported action")
 
-    def cleanup_on_provider(self):
+    def delete_from_provider(self):
```

Comment:
Understood, but I don't really see the need for such a distinction, since:

1. common.vm.delete_from_provider only has one use in test, where it only runs against openstack and is not expecting some other resources to still be present
2. cleanup_on_provider is only defined for azure

We don't lose anything by changing this to a single function.  If anything I think this would be better as a kwarg to `skip_cleanup`, where the Azure method could check that after calling `super().delete_from_provider`.

Reply:
Yeah, cleanup steps are only defined for azure at the moment, but I do intend to have cleanup handled on all providers. But, this is mostly gonna move into wrapanapi anyways in PR240. So we can go with your kwarg approach for an Entity there if we want to.

Reply:
Cool. I guess part of my thinking is that we should always do cleanup of associated VM/Instance resources when deleting on provider.  Any tests that want to delete on provider and then subsequently use/test the associated resources should directly call the `provider.mgmt` methods to delete, and not use the convenience method.

Reply:
Yup, all these convenience methods that are really just wrapanapi convenience methods are gonna live in wrapanapi soon and get out of integration_tests ;)